HPXMLtoOpenStudio/resources/xmlhelper.rb

```diff
@@ -25,6 +25,30 @@ def self.add_element(parent, element_name, value = nil, datatype = nil, defaulte
     return added
   end
 
+  # Inserts the child element with 'element_name' and sets its value. Returns the
+  # child element.
+  def self.insert_element(parent, element_name, index = 0, value = nil, datatype = nil, defaulted = false)
```

Comment:
This has copy/pasted a lot of code. To prevent redundant code, how about updating the `add_element` method to call this method w/ index=-1? I assume that would work.